src/components/history/HistoryFilters.tsx

```diff
@@ -1,26 +1,83 @@
 import { useAppDispatch, useAppSelector } from "../../redux/hooks";
-import { setCurrentFilter, setWaitingForMyDataFiltering } from "../sidebar/actions/productSlice";
+import { setCurrentFilter, setWaitingForMyDataFiltering, setWaitingForMyDataFilteringReset } from "../sidebar/actions/productSlice";
 import { defaultFilterParameters, defaultSpatialSearchEndDate, defaultSpatialSearchStartDate, inputBounds, parameterOptionValues, rasterResolutionOptions } from "../../constants/rasterParameterConstants";
 import { useState } from "react";
 import { OutputGranuleExtentFlagOptions, OutputSamplingGridType, RasterResolution, Adjust, FilterParameters, FilterAction } from "../../types/historyPageTypes";
 import DatePicker from "react-datepicker";
 import "react-datepicker/dist/react-datepicker.css";
 import { Formik } from "formik";
 
+export const getFilterParameters = (currentFilters: FilterParameters, limit?: number, after?: string): UserProductQueryVariables => {
+    const productVariablesObject: ProductQueryParameters = {}
+    const {cycle, pass, scene, outputGranuleExtentFlag, outputSamplingGridType, startDate, endDate} = currentFilters
+
+    if(typeof limit !== 'undefined') productVariablesObject.limit = limit
+    if(typeof after !== 'undefined') productVariablesObject.after = after
+    if(cycle !== 'none') {
+        productVariablesObject.cycle = parseInt(cycle)
+    }
+    if (pass !== 'none') {
+        productVariablesObject.pass = parseInt(pass)
+    }
+    if (scene !== 'none') {
+        productVariablesObject.scene = parseInt(scene)
+    }
+    if (outputGranuleExtentFlag.length > 0) {
+        let outputGranuleExtentFlagToUse = null
+        if(outputGranuleExtentFlag.includes('128 x 128')) outputGranuleExtentFlagToUse = false
+        if(outputGranuleExtentFlag.includes('256 x 128')) outputGranuleExtentFlagToUse = true
+        if(outputGranuleExtentFlagToUse !== null) productVariablesObject.outputGranuleExtentFlag = outputGranuleExtentFlagToUse
+    }
+    if (outputSamplingGridType.length > 0) {
+        let outputSamplingGridTypeToUse = null
+        if(outputSamplingGridType.includes('UTM')) outputSamplingGridTypeToUse = 'UTM'
+        if(outputSamplingGridType.includes('LAT/LON')) outputSamplingGridTypeToUse = 'GEO'
+        if(outputSamplingGridTypeToUse !== null) productVariablesObject.outputSamplingGridType = outputSamplingGridTypeToUse as GridType
+    }
+
+    // filter [status, rasterResolution, utmZoneAdjust, mgrsBandAdjust] after products gotten. 
+    // TODO: Implement these into this function once the filtering endpoint supports it.
+    if(startDate !== 'none') {
+        productVariablesObject.afterTimestamp = new Date(startDate.setSeconds(0,0)).toISOString().replace('Z','')
+    }
+    if(endDate !== 'none') {
+        productVariablesObject.beforeTimestamp = new Date(endDate.setSeconds(0,0)).toISOString().replace('Z','')
```

Comment:
I was mistaken on the `nanoseconds` comment earlier. It does seem that we can support the decimal value so the `setSeconds` isn't needed. Only the `Z` needs to be removed which is usually at the end of the string (probably can be taken care of with a `.substring` call)

https://github.com/openjdk/jdk/blob/4b4a483b6fe7a6fcfdfe6f68faac29099a64c982/src/java.base/share/classes/java/time/format/DateTimeFormatter.java#L906
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring